pkg/config/autodiscovery/autodiscovery.go

```diff
@@ -77,8 +77,8 @@ func DiscoverComponentsFromEnv() ([]config.ConfigurationProviders, []config.List
 	detectedListeners = append(detectedListeners, config.Listeners{Name: "environment"})
 	detectedListeners = append(detectedListeners, config.Listeners{Name: "static config"})
 
-	// Automatic handling of AD providers/listeners should only run in Core agent.
-	if flavor.GetFlavor() != flavor.DefaultAgent {
+	// Automatic handling of AD providers/listeners should only run in the core or process agent.
+	if flavor.GetFlavor() != flavor.DefaultAgent && flavor.GetFlavor() != flavor.ProcessAgent {
```

Comment:
Nit: should this be switched to a switch statement to help readability?

Reply:
Sorry, maybe I'm misunderstanding, but wouldn't that be much more code since we're doing `&&` here and not `||`?